src/Tizen.NUI.Wearable/src/public/Popup.cs

```diff
@@ -0,0 +1,524 @@
+                    }
+                    else
+                    {
+                        CustomAnimation.ShowAnimation(this);
+                    }
+                    window.AddLayer(layer);
+                    layer.RaiseToTop();
+                }
+                else
+                {
+                    window.AddLayer(layer);
+                    layer.RaiseToTop();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Dismiss Popup.
+        /// </summary>
+        /// <param name="animated">True when using Dismissing event</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void Dismiss(bool animated = true)
+        {
+            if (window != null)
+            {
+                if (animated)
+                {
+                    if (CustomAnimation == null)
+                    {
+                        defaultHideAnimation();
+                    }
+                    else
+                    {
+                        CustomAnimation.HideAnimation(this);
+                    }
+                }
+                else
+                {
+                    window.RemoveLayer(layer);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Set timeout milli-second
+        /// </summary>
+        /// <param name="milliSecond">Automatically dissmissed after milli-second</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void SetTimeout(uint milliSecond)
+        {
+            if (timer == null)
+            {
+                timer = new Timer(milliSecond);
+            }
+            else
+            {
+                timer.Stop();
+            }
+
+            timer.Tick += (s, e) =>
+            {
+                Dismiss();
+                return false;
+            };
+            timer.Start();
+        }
+
+        /// <summary>
+        /// SetContentContainerSize
+        /// </summary>
+        /// <param name="width"></param>
+        /// <param name="height"></param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void SetContentContainerSize(int width, int height)
+        {
+            if(contentContainer != null)
+            {
+                contentContainer.Size = new Size(width, height);
+                wrapContent = false;
+            }
+        }
+
+        /// <summary>
+        /// OnUpdate
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected override void OnUpdate()
+        {
+            if (wrapContent)
+            {
+                contentContainer.WidthSpecification = LayoutParamPolicies.WrapContent;
+                contentContainer.HeightSpecification = LayoutParamPolicies.WrapContent;
+            }
+        }
+
+        /// <summary>
+        /// PopupAnimation
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public interface PopupAnimation
```

Comment:
Button used Extension for animation customization. Which design is better?

Reply:
I think Button's Extension has more functionality and tuning points. but I think popup needs only "show" and "hide" animations. it seems that popup has only simple behaviors of showing and hiding.